From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: gngpp <gngppz@gmail.com>
Date: Thu, 18 Sep 2026 00:00:00 +0800
Subject: [PATCH 12/12] boringssl: skip compiler-generated helpers in the
 symbol audit

Auditing libssl, not just libcrypto, reaches the helper functions the compiler
emits for C++ code: clang's `__clang_call_terminate` for exception cleanup, and
MSVC's array constructor and destructor iterators. They belong to the compiler
runtime rather than BoringSSL, carry no identifier prefixing could rewrite, and
being weak they merge instead of colliding. The surrounding list already skips
their siblings, `__cxa_*`, the unwinder personalities and the RTTI records.

Copyright (c) 2025-2026 0x676e67 <gngppz@gmail.com>

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
---
 util/audit_symbols.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/util/audit_symbols.go b/util/audit_symbols.go
--- a/util/audit_symbols.go
+++ b/util/audit_symbols.go
@@ -59,6 +59,7 @@ var skipWeakSymbols = []*regexp.Regexp{
 	regexp.MustCompile(`^_Zn.*`),                         // operator new()
 	regexp.MustCompile(`^__\w+\.get_pc_thunk\..*`),       // PIC
 	regexp.MustCompile(`^___asan_.*`),                    // AddressSanitizer
+	regexp.MustCompile(`^__clang_call_terminate$`),       // clang C++ EH helper
 	regexp.MustCompile(`^__cxa_.*`),                      // libc++abi
 	regexp.MustCompile(`^__dynamic_cast$`),               // libc++abi
 	regexp.MustCompile(`^__emutls_get_address$`),         // emulated TLS
@@ -76,6 +77,7 @@ var skipWeakSymbols = []*regexp.Regexp{
 	regexp.MustCompile(`^\?\?2.*`),                                      // operator new()
 	regexp.MustCompile(`^\?\?3.*`),                                      // operator delete()
 	regexp.MustCompile(`^\?\?_C\@_.*`),                                  // String literals
+	regexp.MustCompile(`^\?\?_[HIJLMN]@Y.*`),                            // Array ctor/dtor iterators
 	regexp.MustCompile(`^\?memchr(@.*)?$`),                              // memchr()
 	regexp.MustCompile(`^\?strchr(@.*)?$`),                              // strchr()
 	regexp.MustCompile(`^_Check_memory_order$`),                         // std::atomic
